system/jlib/jthread.cpp

```diff
@@ -54,10 +54,13 @@ void addThreadTermFunc(ThreadTermFunc onTerm)
 
 void callThreadTerminationHooks(bool isPooled)
 {
+    std::vector<ThreadTermFunc> keepHooks;
     for (auto hook: threadTermHooks)
     {
-        (*hook)(isPooled);
+        if ((*hook)(isPooled) && isPooled)
+           keepHooks.push_back(hook);
```

Comment:
Would it be better to assert (isPooled) rather than checking it?

Reply:
Not sure it makes a lot of difference. The value is generally irrelevant if isPooled is false, as all the structures concerned are about to disappear.